src/quote/provider/index.ts

```diff
@@ -81,46 +57,48 @@ export class FlashMintQuoteProvider
 {
   constructor(
     private readonly rpcUrl: string,
-    private readonly swapQuoteProvider: SwapQuoteProvider,
-    private readonly swapQuoteProviderV2?: SwapQuoteProviderV2,
+    private readonly swapQuoteProviderV2: SwapQuoteProviderV2,
     private readonly swapQuoteOutputProviderV2?: SwapQuoteProviderV2,
   ) {}
 
   async getQuote(
     request: FlashMintQuoteRequest,
   ): Promise<FlashMintQuote | null> {
+    const { rpcUrl, swapQuoteProviderV2, swapQuoteOutputProviderV2 } = this
     const {
-      rpcUrl,
-      swapQuoteProvider,
-      swapQuoteProviderV2,
-      swapQuoteOutputProviderV2,
-    } = this
-    const provider = getRpcProvider(rpcUrl)
-    const { inputToken, inputTokenAmount, isMinting, outputToken, slippage } =
-      request
+      chainId,
+      inputToken,
+      inputTokenAmount,
+      isMinting,
+      outputToken,
+      slippage,
+    } = request
+
     const indexTokenAmount = BigNumber.from(request.indexTokenAmount)
     const indexToken = isMinting ? outputToken : inputToken
     const inputOutputToken = isMinting ? inputToken : outputToken
-    const network = await provider.getNetwork()
-    const chainId = network.chainId
+
```

Comment:
Removing this might save some ms - especially if the provider would actually have to do a network call. In most cases this would probably not happen tho. So only smol save most likely.

Reply:
Adding `chainId` to the quote request was long planned anyway, so good to make that change now.